proxywasm/hostcall.go

```diff
@@ -128,6 +128,15 @@ func GetHttpRequestBody(start, maxSize int) ([]byte, error) {
 	return ret, types.StatusToError(st)
 }
 
+func SetHttpRequestBody(body []byte) error {
+	if len(body) == 0 {
```

Comment:
I image someone would want to actually delete the entire body, and in that case we need to pass body with `len == 0`. Can we allow these cases?

Reply:
I also created a proposal for `ProxySetBufferBytes` in the `proxytest` package and an example using it for testing. Are you interested in that as well in this MR?

Reply:
Cool, I would like to have them as well in this pull request. Thanks!

Reply:
No problem! I've just added them. Please let me know if any additional changes are needed.